src/Microsoft.SqlTools.Hosting/Hosting/Protocol/MessageDispatcher.cs

```diff
@@ -27,14 +27,22 @@ public class MessageDispatcher
         internal Dictionary<string, Func<Message, MessageWriter, Task>> requestHandlers =
             new Dictionary<string, Func<Message, MessageWriter, Task>>();
 
+        internal Dictionary<string, bool> requestHandlerParallelismMap =
```

Comment:
Instead of having a separate dictionary just to keep track of bools I would suggest using a tuple type in the requestHandlers dictionary

https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/value-tuples

`internal Dictionary<string, (Func<Message, MessageWriter, Task>, bool)> requestHandlers =
            new Dictionary<string, (Func<Message, MessageWriter, Task>, bool)>();`

Reply:
I gave this a try but personally the readability feels worse to me (even with the use of field name properties), I might stick with the additional dict here.